### PR TITLE
Added OpenID Connect as a security option

### PR DESCRIPTION
**Pull request by morungos**
This allows `openid-connect` to be used as an authentication option. It's also refactoring some of core to use a wrapped UserDetails rather than a raw Spring Security one, as the naive Spring Security version doesn't offer usernames for OIDC authentication tokens. 

See #961

### CODE REVIEW

**Comment by n1zea144** [comment]
@morungos Looks like there is a typo in the parent pom.xml (line 417).

File: portal/src/main/webapp/WEB-INF/jsp/global/header.jsp [hunk 1]
```diff
@@ -148,8 +148,8 @@
 
                                             <!-- Added call GlobalProperties to check whether to show the Visualize tab -->
                                             <% if (GlobalProperties.showVisualizeYourDataTab()) { %>
-                                            <li class="internal" style="float:left">
-					                            <a href="visualize_your_data.jsp" float="left">VISUALIZE YOUR DATA</a>
+                                            <li class="internal" style="float:right">
```
Comment (n1zea144) [inline portal/src/main/webapp/WEB-INF/jsp/global/header.jsp]: This branch is slightly out of date - you are clobbering an update to this file.

**Comment by morungos** [comment]
OK, I better pick up on this and bring it into line better

**Comment by n1zea144** [comment]
Thanks, yeah,  I just merged some LDAP support this morning.

**Comment by morungos** [comment]
"Invalid signature file digest for Manifest main attributes" Whut?!!

**Comment by jjgao** [comment]
closing this since it's so long ago. Please feel free to reopen if needed.

**Pull request closed without merging**